### ISSUE TITLE
[a11y] 11.2.1.1 Keyboard

### ISSUE THREAD

**Issue by tschiebel** [owncloud/ios-app#1336]
audit: https://infinite.owncloud.com/s/qSvYucPltEYUhhc

https://github.com/BIK-BITV/BIK-App-Test/tree/main/Pr%C3%BCfschritte/de
https://www.etsi.org/deliver/etsi_en/301500_301599/301549/03.02.01_60/en_301549v030201p.pdf

### Personal Space

- [x] In the test, it was not possible to focus the sort button with the keyboard.
focus with the keyboard.
![image](https://github.com/owncloud/ios-app/assets/132444655/8a7a28f5-4db2-4beb-90ef-17e21a8f430b)

> **Fix:** the sort button can now be reached through `Tab` and `Shift + Tab`.

- [x] It is not possible to focus the "More" buttons using the keyboard.
![image](https://github.com/owncloud/ios-app/assets/132444655/07a86e4d-04a8-499d-8450-8023ef0e4a59)

> **Fix:** the more button can either be triggered by pressing the Return key, or by using the system-provided `Tab + Z` menu when the row is focused

### Menu

- [x] The "Eject" button cannot be focused using the keyboard.
![image](https://github.com/owncloud/ios-app/assets/132444655/5f8fd935-56b2-44ae-aead-f8b9ecebccbc)

> **Fix:** the eject action is now available through the system-provided `Tab + Z` menu

### Shared with me

- [x] It is not possible to use the keyboard to focus the "More" buttons.
![image](https://github.com/owncloud/ios-app/assets/132444655/f283761c-2b7d-4509-b6b8-575237a605f9)

> **Fix:** the more button can either be triggered by pressing the Return key, or by using the system-provided `Tab + Z` menu when the row is focused

- [x] It is not possible to use the keyboard to focus the "Accept" and "Reject" buttons.
"Reject" buttons with the keyboard. 

> **Fix:** the Accept and Reject actions can now be reached using the system-provided `Tab + Z` menu when the row is focused

### Shared with others

- [x] The buttons with the arrow pointing to the right cannot be focused and activated using the keyboard.
and activated with the keyboard. Instead, the entire list element is focused, but it
cannot be activated.
![image](https://github.com/owncloud/ios-app/assets/132444655/20d6f431-838d-43bd-80f3-d15044b8a8ea)

> **Fix:** the reveal button can either be triggered by pressing the right arrow key, or by using the system-provided `Tab + Z` menu when the row is focused

### Image Viewer

- [x] It is not possible to change the elements of the carousel using the keyboard.
using the keyboard. It is only possible using swipe gestures.

> **Fix:**: the elements of the carousel can now be navigated with `Shift + CursorLeft` and `Shift + CursorRight` when the carousel is focused

**Comment by felix-schwarz** [comment]
Useful resources debugging and fixing this:
- Documentation
   - https://developer.apple.com/documentation/uikit/focus-based_navigation/debugging_focus_issues_in_your_app?language=objc
   - https://developer.apple.com/documentation/uikit/uifocusdebugger?language=objc
- WWDC 21 videos:
   - Support Full Keyboard Access in your iOS app: https://developer.apple.com/videos/play/wwdc2021/10120/
   - Focus on iPad keyboard navigation: https://developer.apple.com/videos/play/wwdc2021/10260/

**Comment by felix-schwarz** [comment]
Keyboard navigation is a bad fit for additional actions like "Reveal" and "More" in rows. That functionality is therefore made available via Accessibility Custom Actions, reachable via `Tab + Z` while the respective element is focused.

**Comment by jesmrec** [comment]
All the findings above are fixed

Unfortunately, i found more unexpected appearances of `Drag Item`:

- In  items inside the `Locations` section of `Available Offline` shortcut in sidebar
- In `Search` section of sidebar, every item in `Quick Access`

These two should be the latest ones before finishing 😅
